plume-file-core/src/main/java/com/coreoz/plume/file/services/FileService.java

```diff
@@ -143,12 +143,29 @@ public Optional<FileMetadata> fetchMetadata(String fileUniqueName) {
      */
     public void deleteUnreferenced() throws UncheckedIOException {
         List<String> fileUniqueNamesToDelete = fileMetadataService.findUnreferencedFiles();
+        this.deleteFilesByUniqueName(fileUniqueNamesToDelete);
+        logger.debug("{} unreferenced files deleted", fileUniqueNamesToDelete.size());
+    }
+
+    /**
+     * Delete files from types that longer exists in the {@link FileType} enum and can't be linked to stored files
+     * This method must be use with caution and punctually, as it can lead to errors
+     *
+     * @throws UncheckedIOException is a file could not be deleted. This may happen if the file is still referenced.
```

Comment:
UncheckedIOException is a file could not be deleted. This may happen if file storage service cannot delete the file.